framework/src/linearfvkernels/LinearFVFluxKernel.C

```diff
@@ -50,37 +55,45 @@ LinearFVFluxKernel::addMatrixContribution()
 
     if (hasBlocks(_current_face_info->elemInfo()->subdomain_id()))
     {
-      (*_linear_system.matrix).add(dof_id_elem, dof_id_elem, elem_matrix_contribution);
-      (*_linear_system.matrix).add(dof_id_elem, dof_id_neighbor, neighbor_matrix_contribution);
+      _matrix_contribution(0, 0) = elem_matrix_contribution;
+      _matrix_contribution(0, 1) = neighbor_matrix_contribution;
     }
 
     if (hasBlocks(_current_face_info->neighborInfo()->subdomain_id()))
     {
-      (*_linear_system.matrix).add(dof_id_neighbor, dof_id_elem, -elem_matrix_contribution);
-      (*_linear_system.matrix).add(dof_id_neighbor, dof_id_neighbor, -neighbor_matrix_contribution);
+      _matrix_contribution(1, 0) = -elem_matrix_contribution;
+      _matrix_contribution(1, 1) = -neighbor_matrix_contribution;
     }
+    (*_linear_system.matrix).add_matrix(_matrix_contribution, _dof_indices.get_values());
   }
-  else if (auto * bc_pointer =
-               _var.getBoundaryCondition(*_current_face_info->boundaryIDs().begin()))
+  else if (_current_face_type == FaceInfo::VarFaceNeighbors::ELEM ||
```

Comment:
Yeah this works because the flux kernels that are used are queried from the warehouse knowing the subdomains on both sides of the face. So we won't evaluate a kernel that has nothing to do with these subdomains. There might be some room for optimization here. But I'd like to believe that this is sufficient.